### PR TITLE
feat: initialize doctor with init hook

### PR DESCRIPTION
**Pull request by peternhale**
@W-12647786@
Doctor will init via init hook in plugin-info

This must be merge before:
[sfdx-cli #1004](https://github.com/salesforcecli/sfdx-cli/pull/1004)
[cli #633](https://github.com/salesforcecli/cli/pull/633)

### CODE REVIEW

File: src/hooks/init/init_doctor.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import type { Hook } from '@oclif/core';
+import { Logger } from '@salesforce/core';
+import { VersionCommand } from '@oclif/plugin-version';
+import { Doctor } from '../../doctor';
+
+const log = Logger.childFromRoot('plugin-plugin-info:init_doctor');
```
Comment (shetzel) [inline src/hooks/init/init_doctor.ts:13]: nit: why such a long logger name?  `plugin-info` is useful but `plugin-plugin-info`?

**Review by shetzel — APPROVED** [review]

**Comment by shetzel** [comment]
I think we should have at least 1 NUT to ensure the init hook runs as expected.  Another WI maybe.

File: src/hooks/init/init_doctor.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+const hook: Hook<'init'> = async ({ config }): Promise<void> => {
+  log.debug('init_doctor hook');
+  if (process.argv[2] === 'doctor') {
```
Comment (mdonnalley) [inline src/hooks/init/init_doctor.ts:16]: just FYI, `argv` is passed into the init hook, which already has the command name removed. So you could just do this instead,

```typescript
const hook: Hook<'init'> = async ({ config, argv }): Promise<void> => {
  log.debug('init_doctor hook');
  if (argv[0] === 'doctor') {
   // etc...
  }
}
```

Comment (peternhale) [reply src/hooks/init/init_doctor.ts:16]: @mdonnalley what I am seeing in the debugger for argv for command `doctor -d a/b/c/d -c "sfdx org list"` is 

```
Array(4) [-d,
a/b/c/d,
-c,
sfdx org list]
```
I also see the hook signature includes id, which is set to doctor. I am going to go with `id` parameter.

**Pull request closed without merging**